hammer/library/aws/security_groups.py

```diff
@@ -372,6 +374,29 @@ def __str__(self):
         perms = ", ".join([str(perm) for perm in self.permissions])
         return f"{self.__class__.__name__}(Name={self.name}, Id={self.id}, Permissions=[{perms}])"
 
+    def validate_known_ip_soource(self, source_ip):
+        """
+        
+        :param source_ip: ip address
+        :return: boolean
+        """
+        config = Config()
+        known_ip_sources = config.sg.known_ip_sources
+        source_cidr = ipaddress.ip_network(source_ip)
+
+        for known_ip in known_ip_sources:
+            known_ip_cidr = ipaddress.ip_network(known_ip)
+            if known_ip_cidr == source_cidr:
+                return True
+            elif source_ip.endswith("/32"):
+                for ip in known_ip_cidr:
+                    if str(source_cidr) == str(ipaddress.ip_network(ip)):
+                        return True
+            # ipaddress.subnet_of() function new to Python 3.7. Not available in 3.6
+            """elif source_cidr.subnet_of(known_ip_cidr):
+                return True"""
+        return False
+
     def restriction_status(self, cidr):
```

Comment:
I'm looking on this code https://github.com/dowjones/hammer/blob/dev/hammer/identification/lambdas/sg-issues-identification/describe_sec_grps_unrestricted_access.py#L57, it checks if the group is restricted and doesn't push it to db if it is. Should we do the same for safe groups? Any reason to save them to DB?

Reply:
As per testing it is not storing Safe groups in DDB. For Safe groups it is returning with Restricted status.